### PR TITLE
Support CTAS(Create Table As Select ~) statement

### PR DESCRIPTION
**Pull request by devgony**
# Support CTAS(Create Table As Select ~) statement(#404)
- [x] Add test cases: `src/tests/alter/create_table.rs`
- [x] Add `source` to translate if there is query statements: `src/translate/mod.rs`
- [x] Get `source` from ast: `src/ast/mod.rs`
- [x] Pass `source` to executor: `src/executor/execute.rs`
- [x] Implement CTAS
  - [x] Get source table's schema
  - [x] Create new table with the source schema
  - [x] Insert data from source to new table
  - [x] Clear redundant and repetitive code

### CODE REVIEW

File: src/executor/alter/table.rs [hunk 1]
```diff
@@ -40,17 +73,39 @@ pub async fn create_table<T: Debug, U: GStore<T> + GStoreMut<T>>(
...
+                storage.insert_schema(&schema).await
+            } else {
+                Ok((storage, ()))
+            }
+        }
         Err(e) => {
             return Err((storage, e));
         }
-    };
+    }?;
```
Comment (panarch) [inline src/executor/alter/table.rs:87]:
```rust
let (storage, target_table_name) = get_name(name).try_self(storage)?;               
let schema = (|| async {                                                            
    let target_columns_defs = match source.as_ref().map(AsRef::as_ref) {            
        Some(Query {                                                                
            body: SetExpr::Select(select_query),                                    
            ..                                                                      
        }) => {                                                                     
            let TableFactor::Table {                                                
                name: source_name, ..                                               
            } = &select_query.from.relation;                                        
                                                                                    
            let table_name = get_name(source_name)?;                                
            let schema = storage.fetch_schema(table_name).await?;                   
                                                                                    
            let Schema {                                                            
                column_defs: source_column_defs,                                    
                ..                                                                  
            } = schema.ok_or_else(|| -> Error {                                     
                FetchError::TableNotFound(table_name.to_owned()).into()             
            })?;                                                                    
                                                                                    
            source_column_defs                                                      
        }                                                                           
        _ => column_defs.to_vec(),                                                  
    };                                                                              
                                                                                    
    let schema = Schema {                                                           
        table_name: target_table_name.to_string(),                                  
        column_defs: target_columns_defs,                                           
        indexes: vec![],                                                            
    };                                                                              
                                                                                    
    for column_def in &schema.column_defs {                                         
        validate(column_def)?;                                                      
    }                                                                               
                                                                                    
    match (                                                                         
        storage.fetch_schema(&schema.table_name).await?,                            
        if_not_exists,                                                              
    ) {                                                                             
        (None, _) => Ok(Some(schema)),                                              
        (Some(_), true) => Ok(None),                                                
        (Some(_), false) => {                                                       
            Err(AlterError::TableAlreadyExists(schema.table_name.to_owned()).into())
        }                                                                           
    }                                                                               
})()                                                                                
.await;                                                                             

let (storage, schema) = schema.try_self(storage)?;
let storage = match schema {                      
    Some(schema) => storage                       
        .insert_schema(&schema)                   
        .await                                    
        .map(|(storage, _)| storage)?,            
    None => storage,                              
};                                                
```

File: src/executor/alter/table.rs [hunk 2]
```diff
@@ -40,17 +73,39 @@ pub async fn create_table<T: Debug, U: GStore<T> + GStoreMut<T>>(
...
+                offset: None,
+            };
+            let (storage, rows) = (|| async {
+                select(&storage, &query, None)
+                    .await?
+                    .try_collect::<Vec<_>>()
+                    .await
+            })()
+            .await
+            .try_self(storage)?;
```
Comment (ever0de) [inline src/executor/alter/table.rs:104]: You can simplify it by using `async + scope`

```rust
            let (storage, rows) =
                async { select(&storage, &query, None).await?
                    .try_collect().await }
                    .await
                    .try_self(storage)?;
```

**Review by panarch — CHANGES_REQUESTED** [review]
There's only a very tiny comment.
Everything else looks awesome, thanks a lot 👍

**Review by panarch — APPROVED** [review]
👍 👍 👍

**Pull request merged**